RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs

```diff
@@ -9,198 +9,211 @@
-            _memberName = memberName; // Null is a valid value
-            _resolver = resolver ?? Resolver.Empty;
-            _hash = GetHash();
-            Object = CreateObject();
-            ChangeToken.OnChange(section.GetReloadToken, () => ReloadObject(false));
-        }
-
-        /// <summary>
-        /// Force the underlying object to reload from the current configuration.
-        /// </summary>
-        public void Reload() => ReloadObject(true);
-
-        /// <summary>
-        /// Gets the underlying object.
-        /// </summary>
-        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
-        public TInterface Object { get; private set; }
-
-        /// <summary>
-        /// Occurs immediately before the underlying object is reloaded.
-        /// </summary>
-        public event EventHandler Reloading;
-
-        /// <summary>
-        /// Occurs immediately after the underlying object is reloaded.
-        /// </summary>
-        public event EventHandler Reloaded;
-
-        /// <summary>
-        /// Dispose the underlying object if it implements <see cref="IDisposable"/>.
-        /// </summary>
-        public void Dispose() => (Object as IDisposable)?.Dispose();
-
-        private TInterface CreateObject()
-        {
-            // In order to create the object (and avoid infinite recursion), we need to figure out
-            // the concrete type to create and the config section that defines the value to create.
-            Type concreteType;
-            IConfiguration valueSection;
-
-            // If _section contains a type-specified value, extract the type and use the value sub-section.
-            string typeValue = _section[ConfigurationObjectFactory.TypeKey];
-            if (typeValue != null)
-            {
-                // Throw if the value does not represent a valid Type.
-                concreteType = Type.GetType(typeValue, true);
-
-                if (!typeof(TInterface).GetTypeInfo().IsAssignableFrom(concreteType))
-                    throw Exceptions.ConfigurationSpecifiedTypeIsNotAssignableToTargetType(typeof(TInterface), concreteType);
-
-                valueSection = _section.GetSection(ConfigurationObjectFactory.ValueKey);
-            }
-
-            // If there is a registered default type, use it.
-            else if (ConfigurationObjectFactory.TryGetDefaultType(_defaultTypes, typeof(TInterface), _declaringType, _memberName, out concreteType))
-            {
-                // The value section depends on whether the 'ReloadOnChange' flag is set to true.
-                if (string.Equals(_section[ConfigurationObjectFactory.ReloadOnChangeKey]?.ToLowerInvariant(), "true"))
-                    valueSection = _section.GetSection(ConfigurationObjectFactory.ValueKey);
-                else
-                    valueSection = _section;
-            }
-
-            // Throw if no type can be located.
+   // Given that this class is already public,
```

Comment:
This is what I was saying with the end_of_line .editorconfig setting.  It makes it hard to identify the real changes. We should probably change it to just lf.  Or go full on changing to crllf but then we should probably change every file in the repo in a separate pr.